### PR TITLE
[CH6809] Redirects support + tests

### PR DESCRIPTION
**Pull request by sblaurock**
Added methods to support the following [redirect rule REST endpoints](https://docs.constructor.io/rest-api.html#redirect-rules):

addRedirectRule => POST https://ac.cnstrc.com/v1/redirect_rules
getRedirectRules => GET https://ac.cnstrc.com/v1/redirect_rules
getRedirectRule => GET https://ac.cnstrc.com/v1/redirect_rules/[redirect_rule_id]
modifyRedirectRule => PUT https://ac.cnstrc.com/v1/redirect_rules/[redirect_rule_id]
updateRedirectRule => PATCH https://ac.cnstrc.com/v1/synonym_groups/[redirect_rule_id]
removeRedirectRule => DELETE https://ac.cnstrc.com/v1/synonym_groups/[redirect_rule_id]

52 tests added covering all methods and associated parameters listed above, executing in ~43 seconds.

### CODE REVIEW

File: lib/constructorio.js [hunk 1]
```diff
@@ -423,11 +423,130 @@ ConstructorIO.prototype = {
...
+      json,
+    }, (error, response) => {
+      handleServerResponse(error, response, callback);
+    });
+  },
+
+  /**
+   * @description Partially updates a redirect rule for supplied id.
+   */
+  modifyRedirectRule(params, callback) {
```
Comment (rubikzube) [inline lib/constructorio.js]: Should this be `updateRedirectRule` to correspond with other update functions and the API docs?  It doesn't exactly make sense, but is consistent.

File: lib/constructorio.js [hunk 2]
```diff
@@ -423,11 +423,130 @@ ConstructorIO.prototype = {
...
+      json,
+    }, (error, response) => {
+      handleServerResponse(error, response, callback);
+    });
+  },
+
+  /**
+   * @description Completely updates a redirect rule for supplied id.
+   */
+  setRedirectRule(params, callback) {
```
Comment (rubikzube) [inline lib/constructorio.js]: Should this be `modifyRedirectRule` to align with other functions and the API docs?  It doesn't exactly make sense, but is consistent.

File: lib/constructorio.js (hunk 1 above)
Comment (sblaurock) [reply lib/constructorio.js]: 👍

File: lib/constructorio.js (hunk 2 above)
Comment (sblaurock) [reply lib/constructorio.js]: 👍

**Pull request merged**